benchmarks/benchmarks/sparse_linalg_svds.py

```diff
@@ -24,8 +31,17 @@ def setup(self, k, problem, solver):
                                 "svds_benchmark_files.npz")
         matrices = np.load(datafile, allow_pickle=True)
         self.A = matrices[problem][()]
+        _, s, _ = svd(self.A.toarray(), full_matrices=False)
+        self.all_singular_values = s.sort()
 
     def time_svds(self, k, problem, solver):
-        # consider k = int(np.min(self.A.shape) * k)
-        np.random.seed(0)
-        svds(self.A, k=k, solver=solver)
+        rng = np.random.default_rng(0)
+        if solver == 'svd':
+            _, s, _ = svd(self.A.toarray(), full_matrices=False)
+        else:
+            _, s, _ = svds(self.A, k=k, solver=solver, random_state=rng)
+
+        tol = k * np.prod(self.A.shape) * np.finfo(float).eps
```

Comment:
Would it be possible to move some of this boilerplate up to `setup()` to avoid its inclusion in the overall timings?